tutorials/introductory/patchwork_axes.py

```diff
@@ -0,0 +1,195 @@
+"""
+==============
+Patchwork Axes
+==============
+
+This is inspired by the `patchwork
+<https://github.com/thomasp85/patchwork>`__ library for R.  While we
+do not implement the operator overloading style, we do provide a
+pythonic API for specifying (nested) Axes layouts.  If you need to do
+something that is not clearly expressed using this method, please see
+:ref:`sphx_glr_tutorials_intermediate_gridspec.py`.
+
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def label_axes(ax_dict):
+    for k, ax in ax_dict.items():
+        ax.set_title(k, loc="left")
+    return ax_dict
+
+
+###############################################################################
+# Express String specification
+# ============================
+#
+# Using single leters we can "draw" the axes we want.  The following
+
+
+layout = """
+AAC
+BBC
+"""
+
+###############################################################################
+# will give us 3 axes.  Two will be stacked vertically in the left and
+# one will span the vertical space on the right.  The Axes on the left
+# will be twice as wide as the Axes on the right and the Axes on the
+# right will be twice as tall as Axes on the left.  But, as they say
+# a picture is worth a thousand words:
+
+fig = plt.figure(constrained_layout=True)
+axd = fig.build_grid(layout)
+label_axes(axd)
+
+###############################################################################
+# The returned value from `.Figure.build_grid` is a dictionary mapping the
+# label you set to the axes
+
+print(axd)
+
+###############################################################################
+# If instead we wanted a 2x2 grid on the left and the same span on the right
+axd = plt.figure(constrained_layout=True).build_grid(
+    """
+ABC
+DEC
+"""
+)
+label_axes(axd)
+
+###############################################################################
+# If we do not want to fill in all the spaces with axes you can leave
+# it blank
+
+label_axes(
+    plt.figure(constrained_layout=True).build_grid(
+        """
+ aa
+cBB
+cBB
+ dd
+"""
+    )
+)
```

Comment:
We should be consistent in a tutorial, so I'm against suddenly changing this to being used inline.
```suggestion
axd = plt.figure(constrained_layout=True).build_grid(
    """
 aa
cBB
cBB
 dd
"""
)
label_axes(axd)
```
(repeat for all cases below of doing this.)

This has the added benefit that `axd` is forced to be reassigned and thus is never incorrectly pointing to the last place (which it currently is right now.)

Reply:
And I know it's black, but that extra starting line for `"""` is a waste.

Reply:
Fair enough on the in-line, I was not sure which way was clearer.

The """ being on it's own line is so  that the string looks square instead of 


```
""" aaa
cBB
cBB
 dd"""
```

Reply:
As noted above, if passing the string through inspect.cleandoc, one could write e.g.
```
axd = (plt.figure(constrained_layout=True)
       .build_grid("""aaa
                      cBB
                      cBB
                       dd
                   """)
```
or
```
axd = (plt.figure(constrained_layout=True)
       .build_grid("""
                   aaa
                   cBB
                   cBB
                    dd
                   """)
```
which may read nicer depending on the current indentation.

Reply:
`inpsect.cleandoc` is too agressive (`v = ' '*5 + 'aa\n' +' '*4 + 'bbb'` get turned into `'aa\nbbb'`).  However, this is only a problem if you have the default "empty" space be " ".  If we switch that to a non-whitespace (`'.'` or `'-'`?) then we do not have that problem and a whole bunch of stuff get easier.  I'll see what that looks like...

Reply:
Made this change and it is pretty nice :+1: